Rework getting started for 2016

tas50 opened the pull request:
This is probably the first document that most users read. It's mostly right, but subtly wrong in ways that are going to confuse new users. Several small things can be removed to not confuse early on and some of the better new experiences can be expanded upon.

tas50 committed:
This is confusing in a bulleted list

It’s just not needed here

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Chefspec doesn’t run the chef-client

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove chef-zero from the list of tools included

It’s included, but there’s really no reason to mention it in a getting started document. It’s just powering the tools that people actually care about

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
ChefDK included the Chef CLI not Chef itself

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove definitions from the overview

We document them later, but it’s really not something we should expose new users to

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
A workstation == ChefDK

We need to stress this more. Also stress testing in ChefDK vs. “command line tools”

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Cleanup a very confusing sentence

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Update other tags

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Test Kitchen is the preferred name

At one point the idea was to call it Kitchen.ci, but it’s referred to as Test Kitchen now and we should be consistent here to avoid confusion.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
CherDK is a product. Capitalize it

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Fix tenses and remove reference to node object

No need to bring up node object *yet*

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
We don’t author cookbooks on the Supermarket

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
This note makes no sense for new users

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Clarify where settings are made

We’re talking about chefdk in this paragraph not chef-client.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
This diagram is components and tools

Chef-repo is not a tool. These are the tools and components of your workstation

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Rework the ChefDK tool list

group chef-client/ohai since they’re a pair
clarify we’re talking about the chef cli tool, not the chef-client

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Shared is a better word for what Supermarket does

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 commented:
Needs @davidwrede magic, but content should be good to go

thommay commented on chef_master/source/chef_overview.rst:
```diff
@@ -15,7 +15,7 @@ This diagram shows how you develop, test, and deploy your Chef code.
 .. end_tag
 
-* The workstation is the location from which all of Chef is managed, including :doc:`installing the Chef DK </install_dk>`, :doc:`authoring cookbooks </cookbooks>`, and using tools like :doc:`Kitchen </kitchen>`, `chef-zero (a command-line tool that runs locally as if it were connected to a real Chef server) <https://docs.chef.io/ctl_chef_client.html#run-in-local-mode>`__, command-line tools like ...
+* The workstation is the location from which all of Chef is managed, including :doc:`installing the Chef DK </install_dk>`, :doc:`authoring cookbooks </cookbooks>`, and using tools like :doc:`Test Kitchen </kitchen>`, `chef-zero (a command-line tool that runs locally as if it were connected to a real Chef server) <https://docs.chef.io/ctl_chef_client.html#run-in-local-mode>`__, command-line tools ...
```
I think this sentence needs a lot of love.

thommay replied:
I think:
 * remove "installing the chefdk" (if you want to mention ChefDK it should be full works like the rest of the page) since it's not a task that manages chef, but a prerequisite. 
 * Don't mention chef-zero at all. 
 * End the sentence with "and `chef` (for inter....)", dropping the final ", and resources" since you don't use them and they're not part of a workstation.

tas50 replied:
I nuked most of the sentence

thommay commented on chef_master/source/chef_overview.rst:
```diff
@@ -53,7 +53,7 @@ Chef has the following major components:
        Ruby is the programming language that is the authoring syntax for cookbooks. Most recipes are simple patterns (blocks that define properties and values that map to specific configuration items like packages, files, services, templates, and users). The full power of Ruby is available for when you need a programming language.
 
-       Often, a workstation is configured to use the Chef development kit as the development toolkit. The Chef development kit is a package from Chef that provides an optional (but recommended) set of tooling, including Chef itself, the chef command line tool, Kitchen, ChefSpec, Berkshelf, and more.
+       Often, a workstation is configured to use the Chef Development Kit as the development toolkit. The Chef Development Kit is a package from Chef that provides an optional (but recommended) set of tooling, including Chef itself, the chef command line tool, Kitchen, ChefSpec, Berkshelf, and more.
```
I think we should say `recommended` rather than `optional (but recommended)` here.

tas50 replied:
Fixed

thommay commented on chef_master/source/chef_overview.rst:
```diff
@@ -76,42 +76,40 @@ Chef has the following major components:
 The workstation is the location from which most users do most of their work, including:
 
-* Developing cookbooks and recipes (and authoring them using Ruby syntax and patterns)
+* Developing cookbooks and recipes
```
Developing and testing cookbooks and recipes

thommay commented on chef_master/source/chef_overview.rst:
```diff
@@ -76,42 +76,40 @@ Chef has the following major components:
 .. end_tag
 
-While Chef includes tooling like the Chef development kit, encourages integration and unit testing, and defines workflow around cookbook authoring and policy, it's important to note that you know best about how your infrastructure should be put together. Therefore, Chef makes as few decisions on its own as possible. When a decision must be made, the chef-client uses a reasonable default setting t ...
+Chef Development Kit tooling encourages integration and unit testing, and defines workflow around cookbook authoring and policy, but it's important to note that you know best about how your infrastructure should be put together. Therefore, Chef makes as few decisions on its own as possible. When a decision must be made tools uses a reasonable default setting that can be easily changed. While Chef ...
```
The Chef Development Kit tooling

tas50 replied:
Fixed

Commit:
Replicating shared content changes

Signed-off-by: David Wrede <dwrede@chef.io>

tas50 committed:
Updates based on review

I nuked most of the description of the workstation’s role. The sentence was a horrible run on and included details that just don’t fit in an introduction sentence.

Signed-off-by: Tim Smith <tsmith@chef.io>

davidwrede commented:
![gif-keyboard-17298590796998260830](https://cloud.githubusercontent.com/assets/1634752/21165233/fd1a1ec8-c152-11e6-8b37-a8ef16043de3.gif)
